File: app/src/main/java/com/shahryar/cryptoprice/ui/settings/SettingsViewModel.kt
package com.shahryar.cryptoprice.ui.settings

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.*
import com.shahryar.shared.data.CryptoPriceSettings

class SettingsViewModel(private val appSettings: CryptoPriceSettings) :
    ViewModel() {

    var apiKey by mutableStateOf(appSettings.getSetting(CryptoPriceSettings.KEYS.TOKEN))
        private set

    fun onApiKeyChange(value: String) {
        apiKey = value
    }

    fun saveApiKey() {
        apiKey?.let { appSettings.saveSetting(CryptoPriceSettings.KEYS.TOKEN, it) }
    }
}

File: iosApp/iosApp/settings/SettingsViewModel.swift
//
//  SettingsViewModel.swift
//  iosApp
//
//  Created by Shahryar on 5/3/22.
//

import Foundation
import shared

class SettingsViewModel: ObservableObject {
    
    @Published var apiKey: String = CryptoPriceSettings().getSetting(key: "token") ?? ""
}
